src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/TempPE/DesignTimeInputsChangeTracker.cs

```diff
@@ -148,7 +148,12 @@ internal void ProcessFileChangeNotification(IProjectVersionedValue<string[]> arg
         internal void ProcessDataflowChanges(IProjectVersionedValue<ValueTuple<DesignTimeInputs, IProjectSubscriptionUpdate>> input)
         {
             DesignTimeInputs inputs = input.Value.Item1;
-            IProjectChangeDescription configChanges = input.Value.Item2.ProjectChanges[ConfigurationGeneral.SchemaName];
+
+            if (!input.Value.Item2.ProjectChanges.TryGetValue(ConfigurationGeneral.SchemaName, out IProjectChangeDescription configChanges))
```

Comment:
This could be the project going from a correct state -> bad state, in which case your state has changed and you should update your outputs. Otherwise, when the project is fixed the next "diff" is going to be out of sync with what you think the current "state" is.

Use the GetSnapshotOrEmpty to get an empty snapshot.

Reply:
This block works on ProjectChanges, not a snapshot, so GetSnapshotOrEmpty doesn't apply, but you're right this should have been clearing the state.